pkg/core/block/block_test.go

```diff
@@ -218,3 +219,26 @@ func TestBlockCompare(t *testing.T) {
 	assert.Equal(t, 0, b2.Compare(&b2))
 	assert.Equal(t, -1, b2.Compare(&b3))
 }
+
+func TestBlockEncodeDecode(t *testing.T) {
+	t.Run("positive", func(t *testing.T) {
+		b := newDumbBlock()
+		b.Transactions = []*transaction.Transaction{}
+		_ = b.Hash()
+		testserdes.EncodeDecodeBinary(t, b, new(Block))
+	})
+
+	t.Run("bad contents count", func(t *testing.T) {
+		b := newDumbBlock()
+		b.Transactions = make([]*transaction.Transaction, MaxContentsPerBlock)
+		for i := range b.Transactions {
+			b.Transactions[i] = &transaction.Transaction{
+				Script: []byte("my_pretty_script"),
+			}
+		}
+		_ = b.Hash()
+		data, err := testserdes.EncodeBinary(b)
+		require.NoError(t, err)
+		require.True(t, errors.Is(ErrMaxContentsPerBlock, testserdes.DecodeBinary(data, new(Block))))
```

Comment:
This should be the other way `errors.Is(testserdes...., ErrMaxContentsPerBlock)`